### PR TITLE
Moved Manga chapters over to Decimals for fractional releases

### PR DESCRIPTION
**Pull request by Jacob-Tate**
Modified front end to properly display fractional releases 
Modified front end to properly mark fractional releases as read 
Modified front end to properly mark all prior chapters as read

I have tested and marking completed is working.

I changed the functionality of marking all prior chapters as read since I started using the integration after the fact my lastseenchapter was 50 and I wanted to mark 1-49 but couldnt because of the implementation so it now will mark all previous chapters which may have been missed.

NOTE: I am unaware of MAL or MangaUpdates storing any fractional releases since they dont track individual chapters and the APIs return whole numbers so we will continue to receive i32/i64 values and convert them the decimal after import.

NOTE: This does not handle marking prior fractional releases as read only whole numbers since we have no way of knowing they exist.

NOTE: isNumberOrDecimalString exists to bridge the gap with the  previous schema using numbers. Due to the way the storage is occurring the previous manga entries are numbers whereas the new entries are technically strings (even though they are imported as decimal).

### CODE REVIEW

File: apps/frontend/app/routes/_dashboard.media.item.$id._index.tsx [hunk 1]
```diff
@@ -1352,6 +1352,11 @@ const HistoryItem = (props: { history: History; index: number }) => {
 						e.episodeNumber === props.history.showExtraInformation?.episode,
 				)
 		: null;
+	const isNumberOrDecimalString = (value: unknown): boolean =>
+		typeof value === "number" ||
```
Comment (IgnisDa) [inline apps/frontend/app/routes/_dashboard.media.item.$id._index.tsx]: use `isNumber`, `isString` from `@ryot/ts-utils`.

File: apps/frontend/app/routes/_dashboard.media.item.$id._index.tsx [hunk 2]
```diff
@@ -1352,6 +1352,11 @@ const HistoryItem = (props: { history: History; index: number }) => {
...
+		(typeof value === "string" &&
+			!Number.isNaN(Number.parseFloat(value)) &&
```
Comment (IgnisDa) [inline apps/frontend/app/routes/_dashboard.media.item.$id._index.tsx]: re-export `isNan` and `isFinite` from lodash from `@ryot/ts-utils` and use here.

File: apps/frontend/app/routes/_dashboard.media.item.$id._index.tsx [hunk 3]
```diff
@@ -1368,13 +1373,21 @@ const HistoryItem = (props: { history: History; index: number }) => {
...
+	const displayMangaExtraInformation = (() => {
+		const { chapter, volume } = props.history.mangaExtraInformation || {};
+
+		if (isNumberOrDecimalString(chapter)) {
+			const chapterNum = Number.parseFloat(chapter);
+			const isWholeNumber = chapterNum % 1 === 0;
+			return `CH-${isWholeNumber ? Math.floor(chapterNum) : chapterNum}`;
+		}
+
+		if (isNumber(volume)) {
```
Comment (IgnisDa) [inline apps/frontend/app/routes/_dashboard.media.item.$id._index.tsx]: Could you remove the braces?

**Review by IgnisDa — CHANGES_REQUESTED** [review]
LGTM. Most superficial changes.

**Review by IgnisDa — APPROVED** [review]
LGTM. Will merge later today. Thanks for the contribution!

**Review by IgnisDa — CHANGES_REQUESTED** [review]
Please run `moon r frontend:lint` and address the lint errors.

Edit: Looks like biome does not like usage of `isNan` and `isFinite` from lodash. In that case, please revert those changes. Keep the `isNumber` and `isString`. Sorry for the back and forth 😝.

**Comment by Jacob-Tate** [comment]
Should be good now. Let me know if anymore changes are needed.

**Comment by IgnisDa** [comment]
`moon r frontend:typecheck` gives a lot of errors.

**Comment by Jacob-Tate** [comment]
Resolved that

File: libs/ts-utils/src/index.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ import groupBy from "lodash/groupBy";
 import isBoolean from "lodash/isBoolean";
 import isEmpty from "lodash/isEmpty";
 import isEqual from "lodash/isEqual";
+import isInteger from "lodash/isInteger";
```
Comment (IgnisDa) [inline libs/ts-utils/src/index.ts:14]: is biome fine with this?

**Review by IgnisDa — APPROVED** [review]

**Pull request merged**